packages/kbn-object-versioning/lib/content_management_services_versioning.test.ts

```diff
@@ -0,0 +1,403 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { set } from '@kbn/safer-lodash-set';
+import { get } from 'lodash';
+import { getTransforms } from './content_management_services_versioning';
+import type { ServiceDefinitionVersioned } from './content_management_types';
+
+/**
+ * Wrap the key with [] if it is a key from an Array
+ * @param key The object key
+ * @param isArrayItem Flag to indicate if it is the key of an Array
+ */
+const renderKey = (key: string, isArrayItem: boolean): string => (isArrayItem ? `[${key}]` : key);
+
+const flattenObject = (
```

Comment:
You would pass Kibana coding interview 😄 

- https://github.com/elastic/kibana-hiring/blob/master/exercises/flatten_object/flatten_object.md

Reply:
haha 😄